tools/export/gnuarmeclipse/__init__.py

```diff
@@ -30,13 +53,164 @@ class GNUARMEclipse(Exporter):
     TARGETS = [target for target, obj in TARGET_MAP.iteritems()
                if 'GCC_ARM' in obj.supported_toolchains]
 
-    def __generate_uid(self):
-        return "%0.9u" % randint(0, 999999999)
+    @staticmethod
+    def filter_dot(s):
+        """
+        Remove the './' prefix, if present.
+        """
+        if s == None:
+            return None
+        if s[:2] == './':
```

Comment:
This is not compatible with windows paths. You should use "os.path.sep" rather that '/' or '\'